query-primitives/src/main/scala/io/shiftleft/queryprimitives/steps/types/expressions/ControlStructure.scala

```diff
@@ -13,6 +13,9 @@ class ControlStructure[Labels <: HList](raw: GremlinScala.Aux[nodes.ControlStruc
     with ParserTypeNameAccessors[nodes.ControlStructure, Labels]
     with ExpressionBase[nodes.ControlStructure, Labels] {
 
+  val secondChildIndex = new Integer(2)
```

Comment:
Please put the values in a corresponding object to have them static.
I would also suggest a more descriptive name like: trueBranchIndex.
For the `condition` method you missed to introduce a variable